src/Runner.php

```diff
@@ -11,15 +11,24 @@
 class Runner
 {
     protected $index;
+
+    /**
+     * @var MiddlewareStack
+     */
     protected $middleware;
 
-    public function run($middleware, ServerRequestInterface $request, ResponseInterface $response)
+    public function run(MiddlewareStack $middleware, ServerRequestInterface $request, ResponseInterface $response)
```

Comment:
To tell you the truth about this line I was not sure either for the same reason. But some lines below on the invoke function of the same class you are using the get method from the middleware which implies that the property ``$middleware`` is an object and must have the ``get`` function.